### PR TITLE
Add Draft Proposals Document

### PR DESCRIPTION
**Pull request by dave-tucker**
A place for brainstorming the contents of major PR's in to Docker
It's not really libnetwork design, but more how it will be used by
Docker

Signed-off-by: Dave Tucker <dt@docker.com>

### CODE REVIEW

File: docs/proposals.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+### CLI Changes
+
+
+| Command | Flags | Arguments | Example |
+|---------|-------|---------|---------|
+| docker network create | -d, --driver <br> -n, --name | | docker network create -d overlay --name=foo |
+| docker network ls | | | docker network ls |
+| docker network info | | required: network name | docker network info foo |
+| docker network rm | | required: name | docker-network rm foo |
+| docker network join | | required: container id or name <br> required: network name | docker network join aabbccdd1122 foo |
```
Comment (squaremo) [inline docs/proposals.md]: I thought the idea was to have endpoints live independently of being associated with a container?

Comment (dave-tucker) [inline docs/proposals.md]: I think that is a longer term goal. At least for 1.7, I don't think we should allow users to create|destroy endpoints directly.

File: docs/proposals.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+< todo >
+
+## Add "Network" as a first class object in Docker
+
+### CLI Changes
+
+
+| Command | Flags | Arguments | Example |
+|---------|-------|---------|---------|
+| docker network create | -d, --driver <br> -n, --name | | docker network create -d overlay --name=foo |
```
Comment (mapuri) [inline docs/proposals.md]: We need an option to pass config/labels in the cli/api that libnetwork can handover to the drivers.

Example:
docker network create --driver myDriver --name myNet --label prop1=val1 --label prop2=val2

File: docs/proposals.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+| docker network ls | | | docker network ls |
+| docker network info | | required: network name | docker network info foo |
+| docker network rm | | required: name | docker-network rm foo |
+| docker network join | | required: container id or name <br> required: network name | docker network join aabbccdd1122 foo |
+| docker network leave | | required: container id or name <br> required: network name | docker network leave aabbccdd1122 foo |
+| docker network endpoint ls | | | docker network endpoint ls |
+| docker network endpoint info | | required: endpoint id | docker network endpoint info veth01234567 |
```
Comment (mapuri) [inline docs/proposals.md]: Can this take a container-id as well, to return info about all endpoints for a container?

File: docs/proposals.md (hunk 2 above)
Comment (mavenugo) [inline docs/proposals.md]: @mapuri yes. addressed via https://github.com/docker/libnetwork/issues/42

File: docs/proposals.md (hunk 3 above)
Comment (mavenugo) [inline docs/proposals.md]: @mapuri I think we should keep the container -> endpoints info the application using libnetwork (like Docker engine). and we should be able to get that info in `docker inspect <container-id>`

Comment (mapuri) [inline docs/proposals.md]: sure, that shall work as well. So `docker inspect <>` shall give the endpoint-ids for a container and then user can query the detailed info for those endpoints using `docker network endpoint info <>`, is that correct?

Also while we are at this `veth01234567` in the example should really be the endpoint-id generated/allocated by libnetwork and not the netdevice name generated by driver, or is it both?

**Comment by dave-tucker** [comment]
Closing as docker/libnetwirk have now been integrated

**Pull request closed without merging**